plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -536,10 +568,22 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
                 handle.getSchema(),
                 handle.getIndex(),
                 newDomain,
+                newRegexes,
                 handle.getQuery(),
                 handle.getLimit());
 
-        return Optional.of(new ConstraintApplicationResult<>(handle, TupleDomain.withColumnDomains(unsupported), false));
+        return Optional.of(new ConstraintApplicationResult<>(handle, TupleDomain.withColumnDomains(unsupported), newExpression, false));
+    }
+
+    private String likeToRegexp(String like)
+    {
+        String regexp = like.replaceAll("\\\\", "\\\\\\\\"); // first, escape regexp's escape character
+        for (char c : REGEXP_RESERVED_CHARACTERS) {
+            regexp = regexp.replaceAll("\\" + c, "\\\\" + c);
```

Comment:
BTW it would be more efficient to convert LIKE pattern into regex using simple state machine and iterating over characters. see https://github.com/trinodb/trino/blob/ddea70ad46eef4f835dc887036f4a96bec3b2976/core/trino-main/src/main/java/io/trino/type/LikeFunctions.java#L190-L233

(note: the code there escapes only few characters, because it uses Joni with special configuration, where e.g. braces and brackets don't have special meaning)


i don't request a change -- you can still do the transformation using `.replaceAll` -- but worth adding a TODO comment

Reply:
Done